packages/cursorless-engine/src/actions/Remove.ts

```diff
@@ -28,28 +30,41 @@ export default class Delete implements SimpleAction {
     }
 
     const thatTargets = flatten(
-      await runOnTargetsForEachEditor(targets, async (editor, targets) => {
-        const edits = targets.map((target) => target.constructRemovalEdit());
-        const ranges = edits.map((edit) => edit.range);
-
-        const [updatedRanges] = await performEditsAndUpdateRanges(
-          this.rangeUpdater,
-          ide().getEditableTextEditor(editor),
-          edits,
-          [ranges],
-        );
-
-        return zip(targets, updatedRanges).map(
-          ([target, range]) =>
-            new RawSelectionTarget({
-              editor: target!.editor,
-              isReversed: target!.isReversed,
-              contentRange: range!,
-            }),
-        );
-      }),
+      await runOnTargetsForEachEditor(targets, this.runForEditor),
     );
 
     return { thatTargets };
   }
+
+  private async runForEditor(editor: TextEditor, targets: Target[]) {
+    const edits = targets.map((target) => target.constructRemovalEdit());
+
+    const cursorSelections = editor.selections;
+    const editSelections = edits.map(
+      ({ range }) => new Selection(range.start, range.end),
+    );
+    const editableEditor = ide().getEditableTextEditor(editor);
+
+    const [updatedEditorSelections, updatedEditSelections]: Selection[][] =
```

Comment:
I think I'd use `updatedCursorSelections` for consistency

Reply:
done